### PR TITLE
Add an SOW document for Beyond 09

### PR DESCRIPTION
**Pull request by ifireball**

### CODE REVIEW

File: Beyond09-SOW.md [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+
+
+### Demo 2 (April 23rd): Data model and business logic
+
+The purpose of this demo is to show your application's data model and data
+manipulation logic.
+
+Technical requirements
+
+1. For each feature you're planning to add to your project should include:
```
Comment (yftacherzog) [inline Beyond09-SOW.md:126]:
```suggestion
1. Each feature you're planning to add to your project should include:
```

File: Beyond09-SOW.md [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+
+    1. Data model classes
+    2. Migration files for initializing the data model
+    3. Migration files for creating test data for your DEV environment
+    4. Data model functions and model manager classes for the more complex data 
+       operation you expect to do in your applications
```
Comment (yftacherzog) [inline Beyond09-SOW.md:132]:
```suggestion
       operations you expect to do in your applications
```

File: Beyond09-SOW.md [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+    1. Present all design aspects of the features you're working on
+       including [ERDs](https://www.lucidchart.com/pages/how-to-draw-ERD)
+       and [DFDs](https://www.lucidchart.com/pages/data-flow-diagram) (data
+       design and flow documents).
+    2. Show your data model via the Django admin UI
+    3. Present your data manipulation functions via the Django shell
+    4. Show your test code and explain what you've tested
+5. Please try to avoid displaying complex code or complex drawings in your
+   presentation, or repeating the same ideas in different forms. Your challenge
+   it do explian the complex ideas you have about your app, how it should work
```
Comment (yftacherzog) [inline Beyond09-SOW.md:148]:
```suggestion
   is to explain the complex ideas you have about your app, how it should work
```

File: Beyond09-SOW.md [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+   and how it should be used in a simple way that would be understandable to
+   you audience.
```
Comment (yftacherzog) [inline Beyond09-SOW.md:150]:
```suggestion
   your audience.
```

File: Beyond09-SOW.md [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+
+    1. Query data from your database and present it to your users
+    2. Use forms to gather user input
+    3. Store or modify information in the database as a response to user input
+
+2. Implement automated tests for the features you've completed. The tests 
+   should cover all aspects of the feature's functionality including failure 
+   scenarios, handling of invalid data and unauthorized access scenarios. 
+
+**Demo contents**
```
Comment (yftacherzog) [inline Beyond09-SOW.md:196]:
```suggestion
**Demo content**
```

File: Beyond09-SOW.md [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+
+After the first 2 weeks, you'll be divided into teams - each team with 4-6 students.
+
+Each team will have the following mentors:
+
+* Team 1: Yftach Herzog, Yuval Vinokur, Nitzan Mordechai, Sebastian Scheinkman
+* Team 2: Kobi Hakimi, Moshe Sheena, Or Mergi, Ygal Blum
+* Team 3: Yariv Rachmani, Benny Rochwerger, Amos Mastbaum, Daniel Keler
+* Team 4: Nati Friedman, Omer Amsalem, Noam Finkelstein, Alicia Petrunin, Galit Rosenthal
+
```
Comment (kobihk) [inline Beyond09-SOW.md:236]: Maybe it will be better to move these lines to line # 30 to do it in a chronological way

File: Beyond09-SOW.md [hunk 7]
```diff
@@ -0,0 +1,508 @@
...
+## Project ideas
+
+You can consider below ideas or think up your own.
```
Comment (yftacherzog) [inline Beyond09-SOW.md:239]:
```suggestion
You can consider below ideas or think of your own.
```

File: Beyond09-SOW.md [hunk 8]
```diff
@@ -0,0 +1,508 @@
...
+
+* Dating website
+* Drop shipping web store - selling other people's products on your website
+* Personal password maintenance application
+* Course certificate generation and distribution platform
+* An online riddle or math game
+
+Please avoid the following ideas as they have already been implemented too many
+time in the past by Beyond students:
```
Comment (yftacherzog) [inline Beyond09-SOW.md:248]:
```suggestion
times in the past by Beyond students:
```

File: Beyond09-SOW.md [hunk 9]
```diff
@@ -0,0 +1,508 @@
...
+
+* Platform for finding part-time jobs for students.
+* Platform for academic courses and grading.
+* Platform for finding/recruiting jobs/employess.
```
Comment (yftacherzog) [inline Beyond09-SOW.md:252]:
```suggestion
* Platform for finding/recruiting jobs/employees.
```

File: Beyond09-SOW.md [hunk 10]
```diff
@@ -0,0 +1,508 @@
...
+* Platform for for finding roomates or appartments.
```
Comment (yftacherzog) [inline Beyond09-SOW.md:253]:
```suggestion
* Platform for finding roommates or apartments.
```

File: Beyond09-SOW.md [hunk 11]
```diff
@@ -0,0 +1,508 @@
...
+* Platform for freelancers to offer their services.
+* P2P Lending service
+
+__Note:__ The project used as a learning platform, it should not be
```
Comment (yftacherzog) [inline Beyond09-SOW.md:257]:
```suggestion
__Note:__ The project should be used as a learning platform, it should not be
```

File: Beyond09-SOW.md [hunk 12]
```diff
@@ -0,0 +1,508 @@
...
+ unique or amazing, it could defeat learning process and goals.
```
Comment (yftacherzog) [inline Beyond09-SOW.md:258]:
```suggestion
 unique or amazing. This could defeat learning process and goals.
```

File: Beyond09-SOW.md [hunk 13]
```diff
@@ -0,0 +1,508 @@
...
+
+## Syllabus (list of sessions)
+
+| #    | Date                | Topic(s)                           | Lecturer        | Location        |
+|------|---------------------|------------------------------------|-----------------|-----------------|
+|-     |Wed, Feb 19, 2023    |First challenge hand-off and student selection (30m pre-course session)|Barak Korren|Online/Remote|
```
Comment (yftacherzog) [inline Beyond09-SOW.md:264]: What does it mean student selection in the context of this meeting. Are students been selected there?

File: Beyond09-SOW.md [hunk 14]
```diff
@@ -0,0 +1,508 @@
...
+|3     |Sun, Mar 19, 2023    |Databases, data model and the ORM   |Omer Amsalem     |Online/Remote    |
+|      |                     |Project teams announcement          |                 |                 |
+|**4** |**Sun, Mar 26, 2023**|**Demo 1: Development environment** |Barak            |Online/Remote    |
+|-     |Sun, Apr 02, 2023    |_(Passover - College is closed)_    |-                |                 |
+|-     |Sun, Apr 09, 2023    |_(Passover - College is closed)_    |-                |                 |
+|5     |Sun, Apr 16, 2023    |Business logic and Testing          |Eddy Haas        |Online/Remote    |
+|**6** |**Sun, Apr 23, 2023**|**Demo 2: Data Model and business logic**|**TBD**     |**Online/Remote**|
+|-     |**Apr 24 - Apr 30**  |**Performance review week**         |**-**            |**-**            |
+|7     |Sun, Apr 30, 2023    |UI: Routes, Views and Templates     |Yariv Rachmani   |Online/Remote    |
+|8     |Sun, May 07, 2023    |Frontend, FE, Intro: FE tecnologies |Benny Powers     |Online/Remote    |
```
Comment (yftacherzog) [inline Beyond09-SOW.md:280]:
```suggestion
|8     |Sun, May 07, 2023    |Frontend (FE), Intro: FE technologies |Benny Powers     |Online/Remote    |
```

File: Beyond09-SOW.md (hunk 13 above)
Comment (kobihk) [inline Beyond09-SOW.md:264]:
```suggestion
|-     |Sun, Feb 19, 2023    |First challenge hand-off and student selection (30m pre-course session)|Barak Korren|Online/Remote|
```

File: Beyond09-SOW.md [hunk 15]
```diff
@@ -0,0 +1,508 @@
...
+* The aim, focus and completeness - A good PR does one thing, does it well and is
+  self-contained - single responsibility principle. It does not break existing
+  functionality.
+* Description quality - a PR description should explain what it changes, and 
+  why. It should also link to a relevant ticket that contains the user story 
+  that the PR implements.
+
+    1. If it's a frontend change it should contain also screenshots
+
+* Commit quality - a PR should have at least 1 commit where each one:
```
Comment (yftacherzog) [inline Beyond09-SOW.md:321]:
```suggestion
* Commit quality - each commit in a PR:
```

File: Beyond09-SOW.md [hunk 16]
```diff
@@ -0,0 +1,508 @@
...
+
+    1. Meet the project's coding guidelines (See more about coding guidelines in
+       the "Project documentation" section below)
+    2. Be easy to read and understand
+    3. [Have relevant comments](https://itnext.io/what-makes-a-good-code-comment-5267debd2c24)
+
+* Discussion and review
+
+    1. A PR must be reviewed and approved by other team members
+    2. We expect to see discussion that indicated the team members understand
```
Comment (yftacherzog) [inline Beyond09-SOW.md:345]:
```suggestion
    2. We expect to see discussion that indicates the team members understand
```

File: Beyond09-SOW.md [hunk 17]
```diff
@@ -0,0 +1,508 @@
...
+    * Must be a list of measurable achievements that indicate the work in the
+      issue is done
+
+* Status Labels (Optional)
+
+    1. Each issue can be labeled to indicate its status as:
+
+        * Status/New - An issue that was just created, you can decide that
+          unlabeled issues are considered new
+        * Status/Backlog - A issue that was discussed and accepted for work
```
Comment (yftacherzog) [inline Beyond09-SOW.md:382]:
```suggestion
        * Status/Backlog - An issue that was discussed and accepted for work
```

File: Beyond09-SOW.md [hunk 18]
```diff
@@ -0,0 +1,508 @@
...
+       between issues to the code changes that are made to resolve them
+    2. When a PR is indicated to fix an issue - GitHub will close the issue when
+       the PR is merged. We expect most issues to be closed using this automated
+       mechanism.
+
+## Guidelines for writing and maintaining project documentation
+
+* README.md file
+
+    1. Serves as the landing page for your project in GitHub- should tell new
```
Comment (yftacherzog) [inline Beyond09-SOW.md:411]:
```suggestion
    1. Serves as the landing page for your project in GitHub - should tell new
```

File: Beyond09-SOW.md [hunk 19]
```diff
@@ -0,0 +1,508 @@
...
+       users and contributors about your project and where to find more
+       information about to use it or contribute to it. When your project is
```
Comment (yftacherzog) [inline Beyond09-SOW.md:413]:
```suggestion
       information about how to use it or contribute to it. When your project is
```

File: Beyond09-SOW.md [hunk 20]
```diff
@@ -0,0 +1,508 @@
...
+|**-** |**Fri, Mar 10, 2023**|**Second challenge submission date**|**_(Self work)_**|                 |
+|2     |Sun, Mar 12, 2023    |Introduction to Python              |Barak Korren     |Online/Remote    |
+|3     |Sun, Mar 19, 2023    |Databases, data model and the ORM   |Omer Amsalem     |Online/Remote    |
+|      |                     |Project teams announcement          |                 |                 |
+|**4** |**Sun, Mar 26, 2023**|**Demo 1: Development environment** |Barak            |Online/Remote    |
+|-     |Sun, Apr 02, 2023    |_(Passover - College is closed)_    |-                |                 |
+|-     |Sun, Apr 09, 2023    |_(Passover - College is closed)_    |-                |                 |
+|5     |Sun, Apr 16, 2023    |Business logic and Testing          |Eddy Haas        |Online/Remote    |
+|**6** |**Sun, Apr 23, 2023**|**Demo 2: Data Model and business logic**|**TBD**     |**Online/Remote**|
+|-     |**Apr 24 - Apr 30**  |**Performance review week**         |**-**            |**-**            |
```
Comment (kobihk) [inline Beyond09-SOW.md:278]: Just take into account that we have:
25.4 - Yom HaZikaron
26.4 - Yom HaAtzmaut

File: Beyond09-SOW.md [hunk 21]
```diff
@@ -0,0 +1,508 @@
...
+       small you can have all the documentation included in the README, but as
+       it grows we expect different documentation sections to move to their own
+       files.
+
+* User Guide (Optional - the famous Achilles Heel of Open Source)
+
+    1. Documents all aspects of using the project from installation to daily
+       operation. You might have different guides to different types of users -
+       for example you may have an administration guide targeted at system
+       administrators and an end-user guide that is meant for simple end users.
```
Comment (yftacherzog) [inline Beyond09-SOW.md:423]:
```suggestion
       administrators and an end-user guide that is meant for regular end users.
```

File: Beyond09-SOW.md [hunk 22]
```diff
@@ -0,0 +1,508 @@
...
+
+* Contribution guide - Meant to help new developers learn how to contribute to
+  your project. Should include the following sections:
+
+    1. Development process - You process and conventions for managing tickets,
```
Comment (yftacherzog) [inline Beyond09-SOW.md:428]:
```suggestion
    1. Development process - Your process and conventions for managing tickets,
```

File: Beyond09-SOW.md [hunk 23]
```diff
@@ -0,0 +1,508 @@
...
+       what users and contributors are legally allowed to do.
+    2. In the open source community, developers typically pick an
+       [OSI-approved](https://opensource.org/licenses) license for their  
+       projects.
+    3. In Beyond, the mentors will create the project repo for you with the 
+       permissive [MIT license](https://opensource.org/licenses/MIT). We require
+       that you keep this license for the Beyond project in order to facilitate
+       sharing with other Beyond students and the Red Hat team.
+
+## System Requirements
```
Comment (yftacherzog) [inline Beyond09-SOW.md:460]:
```suggestion
## System Requirements (student workstations) 
```

File: Beyond09-SOW.md [hunk 24]
```diff
@@ -0,0 +1,508 @@
...
+  Slack.
+* Ask for code reviews by mentioning the mentors in the PR
+* Mention people with "@" in GitHub issues and PRs when you want them to comment
+  or review.
+* Link your college email address with your GitHub account, so we can find you by
+  the address
+
+## Buzz words
+
+* PR - A pull request (PR) is a method of submitting contributions to an open
```
Comment (yftacherzog) [inline Beyond09-SOW.md:484]:
```suggestion
* PR - A Pull Request is a method of submitting contributions to an open
```

**Review by yftacherzog — APPROVED** [review]

File: Beyond09-SOW.md (hunk 3 above)
Comment (kobihk) [inline Beyond09-SOW.md:148]:
```suggestion
   it does explain the complex ideas you have about your app, how it should work
```

File: Beyond09-SOW.md (hunk 4 above)
Comment (kobihk) [inline Beyond09-SOW.md:150]:
```suggestion
   your audience.
```

File: Beyond09-SOW.md (hunk 10 above)
Comment (kobihk) [inline Beyond09-SOW.md:253]:
```suggestion
* Platform for finding roommates or apartments.
```

File: Beyond09-SOW.md (hunk 9 above)
Comment (kobihk) [inline Beyond09-SOW.md:252]:
```suggestion
* Platform for finding/recruiting jobs/employees.
```

File: Beyond09-SOW.md [hunk 25]
```diff
@@ -0,0 +1,508 @@
...
+  * "Does the Red Hat team know who you are?"
+* **Collaboration**
+  * Does the student collaborate meaningfully with their teammates and 
+    mentors via the various monitored communication channels
+  * Does the student create PRs that are easy to understand and review
+  * Does the student participate meaningfully in the PR review process
+* **Execution**
+  * Are tasks done on time
+  * Is work being tracked and allocated effectively within the team
+  * Does code meet the quality requirements defined by the mentors
```
Comment (kobihk) [inline Beyond09-SOW.md:304]:
```suggestion
  * Does the code meet the quality requirements defined by the mentors
```

**Review by kobihk — CHANGES_REQUESTED** [review]

File: Beyond09-SOW.md [hunk 26]
```diff
@@ -0,0 +1,508 @@
...
+3. Present project landing page
+4. Work agreements
+    * Contributing doc.
+    * Meetings routines and participants list. 
+    * Communications channels with team mates and mentors.
+
+
+### Demo 2 (April 23rd): Data model and business logic
+
+The purpose of this demo is to show your application's data model and data
```
Comment (Yarboa) [inline Beyond09-SOW.md:121]:
```suggestion
Demo goal, show your application's data model and data
```

File: Beyond09-SOW.md [hunk 27]
```diff
@@ -0,0 +1,508 @@
...
+    4. Show your test code and explain what you've tested
+5. Please try to avoid displaying complex code or complex drawings in your
+   presentation, or repeating the same ideas in different forms. Your challenge
+   it do explian the complex ideas you have about your app, how it should work
+   and how it should be used in a simple way that would be understandable to
+   you audience.
+
+**Work division**
+
+It is recommended that you divide the work in your team in a "vertical" way -
```
Comment (Yarboa) [inline Beyond09-SOW.md:154]:
```suggestion
It is recommended to divide the work among teammates in a "vertical" way -
```

File: Beyond09-SOW.md [hunk 28]
```diff
@@ -0,0 +1,508 @@
...
+this means that each team member will have one or more project features assigned
```
Comment (Yarboa) [inline Beyond09-SOW.md:155]:
```suggestion
this means that each teammate will have one or more project features assigned
```

File: Beyond09-SOW.md [hunk 29]
```diff
@@ -0,0 +1,508 @@
...
+to them, and they will be responsible for implementing all aspects of said
+features including the data model, the tests, the control logic and the UI. 
+A technical way to achieve this in Django is to have each team member create
```
Comment (Yarboa) [inline Beyond09-SOW.md:158]:
```suggestion
A technical way to achieve it in Django, each team member create 
```

File: Beyond09-SOW.md [hunk 30]
```diff
@@ -0,0 +1,508 @@
...
+**Work division**
+
+It is recommended that you divide the work in your team in a "vertical" way -
+this means that each team member will have one or more project features assigned
+to them, and they will be responsible for implementing all aspects of said
+features including the data model, the tests, the control logic and the UI. 
+A technical way to achieve this in Django is to have each team member create 
+and maintain one or more separate Django "apps". 
+
+Working in this way will provide several benefits:
```
Comment (Yarboa) [inline Beyond09-SOW.md:161]:
```suggestion
Working as recommended will provide several benefits:
```

File: Beyond09-SOW.md [hunk 31]
```diff
@@ -0,0 +1,508 @@
...
+
+1. It will allow each team member to experience all the layers of the technical
+   stack
+2. It can reduce the amount of Git conflicts you may encounter as you work on
+   your projects because different features can live in different files.
+
+### Performance Review (between April 24th to April 30) :
+
+The student team mentors will set up a personal 10-minute meeting with each of
```
Comment (Yarboa) [inline Beyond09-SOW.md:170]:
```suggestion
Group's mentors will set up a personal 10-minute meeting with each of
```

File: Beyond09-SOW.md [hunk 32]
```diff
@@ -0,0 +1,508 @@
...
+the students in their team and give them feedback about their performance in the
+course so far: What did they do well and where should they improve. As a student
+you should take the time to think about how you perceive your own performance,
+how is that different from how the mentors see it and why. Take the opportunity
+to learn how to improve!
+
+### Demo 3 (May 21st): Control logic and UI
+
+The purpose of this demo is to review your ability to implement the application
```
Comment (Yarboa) [inline Beyond09-SOW.md:179]:
```suggestion
Demo goal, review your ability to implement the application
```

File: Beyond09-SOW.md [hunk 33]
```diff
@@ -0,0 +1,508 @@
...
+Note that all these topics have been covered by the introduction exercise you
+have been given.
+
+**Work division**
+
+Ensure that every member of your team plays an active role in preparing for
+the demo, we recommend that you divide the work between you in the following way:
+
+1. Student1: vagrant
+2. Student2: bootstrap.sh
```
[the remark this answers is not in the document]
Comment (ifireball) [reply Beyond09-SOW.md:73]: Done.

File: Beyond09-SOW.md [hunk 34]
```diff
@@ -0,0 +1,508 @@
...
+   issues
+6. Make sure you assign your teammates as well as your mentors as reviewers for
+   your PRs
+7. The review process can take a few days - make sure you send your PRs early
+   enough, so they will be merged in time for the demo
+8. You will be given a template for each demo with the specific details that are
+   required
+9. Designate a primary speaker for your team. Other members are encouraged to
+   speak as well, but if so, plan who talks on what so you can all finish within
+   the allotted time (15m).
```
[the remark this answers is not in the document]
Comment (ifireball) [reply Beyond09-SOW.md:101]: Nope. "allotted" is an actual word.

[the remark this answers is not in the document]
Comment (yftacherzog) [reply Beyond09-SOW.md:101]: :bulb:

File: Beyond09-SOW.md (hunk 13 above)
Comment (ifireball) [reply Beyond09-SOW.md:264]: We talk about the selection process

File: Beyond09-SOW.md (hunk 6 above)
Comment (ifireball) [reply Beyond09-SOW.md:236]: done

File: Beyond09-SOW.md (hunk 13 above)
Comment (ifireball) [reply Beyond09-SOW.md]: fixed

File: Beyond09-SOW.md (hunk 3 above)
Comment (ifireball) [reply Beyond09-SOW.md]: fixed to "is to"

**Comment by ifireball** [comment]
@Yarboa , @kobihk I made the fixed you've asked for, please re-review and approve.

**Review by kobihk — APPROVED** [review]

**Comment by kobihk** [comment]
Thanks @ifireball !!

**Pull request merged**